src/backend/distributed/connection/remote_commands.c

```diff
@@ -125,46 +125,68 @@ ReportConnectionError(MultiConnection *connection, int elevel)
 void
 ReportResultError(MultiConnection *connection, PGresult *result, int elevel)
 {
-	char *sqlStateString = PQresultErrorField(result, PG_DIAG_SQLSTATE);
-	char *messagePrimary = PQresultErrorField(result, PG_DIAG_MESSAGE_PRIMARY);
-	char *messageDetail = PQresultErrorField(result, PG_DIAG_MESSAGE_DETAIL);
-	char *messageHint = PQresultErrorField(result, PG_DIAG_MESSAGE_HINT);
-	char *messageContext = PQresultErrorField(result, PG_DIAG_CONTEXT);
+	/* PGresult must be released before leaving this function. */
+	PG_TRY();
+	{
```

Comment:
Well, this is taken directly from pgfdw_report_error (on which the rest of this function also seems to be based), so it seems reasonably battle-tested.